lib/misc/cells.js

```diff
@@ -0,0 +1,64 @@
+(function() {
+  var scopes, values;
+
+  scopes = require('./scopes');
+
+  values = require('underscore-plus').values;
+
+  module.exports = {
+    getRange: function(ed) {
+      var buffer, cursor, end, regex, regexString, start;
+      buffer = ed.getBuffer();
+      start = buffer.getFirstPosition();
+      end = buffer.getEndPosition();
+      regexString = atom.config.get('julia-client.cellDelimiter');
+      regex = new RegExp(regexString);
+      cursor = ed.getCursorBufferPosition();
+      if (cursor.row > 0) {
+        buffer.backwardsScanInRange(regex, [start, cursor], function(arg) {
+          var range;
+          range = arg.range;
+          return start = range.start;
+        });
+      }
+      buffer.scanInRange(regex, [cursor, end], function(arg) {
+        var range;
+        range = arg.range;
+        return end = range.start;
```

Comment:
I think if the range returned by `cells.getRange` is from the start of the first line below the top cell delimiter to the end of the line above the end delimiter, everything will work nicely, without needing changes to Ink. This makes sense because then the range is, I think, the same as if you just select the code in the lines between the cell delimiters.

Funnily enough having the result appear below the delimiter was natural to me (coming from Jupyter), but I think it makes sense in atom to have it above, to avoid problems as you said with overlapping ranges and preserving results.